lldb/source/Utility/Status.cpp

```diff
@@ -224,6 +224,14 @@ const char *Status::AsCString(const char *default_error_str) const {
     if (!m_string.empty() && m_string[m_string.size() - 1] == '\n')
       m_string.pop_back();
 
+  // FIXME: Workaround for ErrorList[ExpressionError, ...].
+  if (m_error.isA<llvm::ErrorList>()) {
+    while (!m_string.empty() && m_string[0] == '\n')
+      m_string = std::string(m_string.data() + 1, m_string.size() - 1);
+    if (!m_string.empty() && m_string[m_string.size() - 1] != '\n')
+      m_string += '\n';
+  }
```

Comment:
It's not clear what this is trying to do. A comment would be nice here.

Reply:
Also, stepping back a bit, I'm wondering if using an ErrorList is the best way to represent these errors. The alternative I'm considering is having a single ExpressionError object which contains a vector of "diagnostics" as a regular member. If nothing else, that would give you more control over how these get serialized into a string (but maybe it would also be easier to print them out using the fancy ascii art?)

Reply:
That's what I had originally, I think I might have misunderstood an earlier comment from you suggesting we represent the expression failure as ErrorInfo to mean that you preferred having them be an ErrorList. I can definitely change that back. It would make iterating over the details a tiny bit less cumbersome.

Reply:
Yeah, I did not mean to imply that. Sorry for the confusion.